tests/__init__.py

```diff
@@ -19,11 +20,13 @@ def load_fixture(filename):
 
 
 def create_pull_files(data):
-    return [PullFile(f) for f in json.loads(data)]
+    session = GitHubSession()
+    return [PullFile(f, session) for f in json.loads(data)]
```

Comment:
Are these constructor changes part of the github3.py upgrade?

Reply:
Yeah, the individual classes like `PullFile` all descend from [github3.models.GitHubCore](https://github.com/sigmavirus24/github3.py/blob/162f247b4664a2612b2cfd959497678b069f2c5e/src/github3/models.py#L29), which expects a session to be passed into `__init__`.

I'm not sure how important it is to have a valid session object in each test case, e.g. PullFile doesn't seem to have any methods that require a valid session, but certain functionality, like constructing URLs, seems to be delegated to the object's session.

An example that has many calls to [GitHubCore._build_url](https://github.com/sigmavirus24/github3.py/blob/162f247b4664a2612b2cfd959497678b069f2c5e/src/github3/models.py#L224-L226) is the [_Repository class](https://github.com/sigmavirus24/github3.py/blob/162f247b4664a2612b2cfd959497678b069f2c5e/src/github3/repos/repo.py#L49), which is the base class for a couple different repository classes.

Reply:
Also, I tried to account for the various changes to tests and some of the "why" in the commit message for https://github.com/markstory/lint-review/pull/233/commits/161d81b27f6e297d654e2714bb245ea834415d91 . I know it's a big patch and some of the changes like the updated json files aren't super easy to review by looking at a diff.